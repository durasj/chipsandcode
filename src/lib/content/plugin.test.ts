import { describe, expect, it } from 'vitest';
import contentPlugin from './plugin';

describe('Markdown Content Vite Plugin', () => {
  it('Has correct structure', () => {
    const plugin = contentPlugin();

    expect(plugin).toMatchObject({
      name: 'transform-markdown-content',
      enforce: 'pre',
      transform: expect.any(Function),
    });
  });

  it('Ignores files that are not .md', () => {
    const plugin = contentPlugin();

    expect(plugin.transform('const x = 123;', 'component.ts')).toBeUndefined();
    expect(plugin.transform('Some Content', 'component.svelte')).toBeUndefined();
    expect(plugin.transform('* { color: #000000 !important; }', 'component.css')).toBeUndefined();
  });

  it('Processes markdown into code', () => {
    const plugin = contentPlugin();

    const markdown = `---
foo: bar
---

Content
`;

    expect(plugin.transform(markdown, 'content.md')).toMatchSnapshot();
  });
});
