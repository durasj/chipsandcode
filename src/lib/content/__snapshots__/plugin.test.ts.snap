// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Markdown Content Vite Plugin > Processes markdown into code 1`] = `
{
  "code": "export default {
	meta: {
		foo: \\"bar\\"
	},
	content: {
		$$mdtype: \\"Tag\\",
		name: \\"article\\",
		attributes: {
		},
		children: [
			{
				$$mdtype: \\"Tag\\",
				name: \\"p\\",
				attributes: {
				},
				children: [
					\\"Content\\"
				]
			}
		]
	}
};",
  "map": null,
}
`;
